### PR TITLE
add project list visual component

### PR DESCRIPTION
**Pull request by drcoolsanjeev**
### Type of change

- Enhancement / new feature

### Description

Add visual component and unit test for project list 
Design - https://marvelapp.com/5gbji22/screen/66859702

image

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [x] Write tests and make sure they pass
- [x] Resolve all warnings
- [x] Update license
- [x] Write test cases for null checks in Unit Tests
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      displayName: "k8s_iot",
+      namespace: "devops_jbosstest1",
+      status: StatusTypes.ACTIVE,
+      creationTimestamp: "2020-05-21T14:40:28.607Z",
+      errorMessageRate: 25,
+      deviceCount: 10500,
+      activeCount: 7100
+    }
+  ];
+  it("should render without crashing", () => {
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.test.tsx:41]: As discussed, please include a test case for null checks.

Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.test.tsx:45]: done

File: console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+import {
+  StatusTypes,
+  ProjectStatus,
+  ProjectError,
+  ProjectEntities,
+  ProjectTypePlan,
+  ProjectTypes
+} from "modules/msg-and-iot/utils";
+
+export const StyleForTable = StyleSheet.create({
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx:29]: don't need to export styles, it should be for specific component only and i would like to suggest variable name as styles.

File: console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  connectionCount?: number;
+  deviceCount?: number;
+  activeCount?: number;
+  selected?: boolean;
+  errorMessages?: Array<string>;
+}
+
+export interface IProjectListProps extends Pick<TableProps, "sortBy"> {
+  onSort?: (_event: any, index: number, direction: SortByDirection) => void;
+  projects: Array<IProject>;
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx:55]: can we please try to use IProject[] instead of Array<IProject>.  IProject[] is more readable.

File: console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  plan?: string;
+}
+
+interface IProjectStatusProps {
+  phase: string;
+  messages?: Array<string>;
+}
+interface IProjectErrorProps {
+  errorCount?: number;
+  errorMessages?: Array<string>;
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx:34]: can we use syntax stringp[]

File: console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+}
+interface IProjectEntitiesProps {
+  projectType: ProjectTypes;
+  addressCount?: number;
+  connectionCount?: number;
+  deviceCount?: number;
+  activeCount?: number;
+}
+interface IProjectErrorMessagesProps {
+  messages?: Array<string>;
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx:44]: can we try syntax stringp[]

File: console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx (hunk 3 above)
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx:55]: can we use syntax IProject[] which is more readable.

File: console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+  }
+});
+interface IProjectTypePlanProps {
+  type: string;
+  plan?: string;
+}
+
+interface IProjectStatusProps {
+  phase: string;
+  messages?: Array<string>;
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx:30]: can we use syntax string[].

File: console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx (hunk 5 above)
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx:44]: can we use syntax string[].

File: console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+  addressCount?: number;
+  connectionCount?: number;
+  deviceCount?: number;
+  activeCount?: number;
+}
+interface IProjectErrorMessagesProps {
+  messages?: Array<string>;
+}
+
+const statusToDisplay = (phase: string) => {
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx:47]: can revisit the function name statusToDisplay.
This function purpose to return icon based on phase. I would like to suggest one name as
getStatusIconByPhase

File: console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+          <ProjectErrorMessages messages={errorMessages} />
+        </>
+      )}
+    </>
+  );
+};
+
+const ProjectEntities: React.FunctionComponent<IProjectEntitiesProps> = ({
+  projectType,
+  addressCount,
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx:108]: one suggestion to set default value if not coming as props, set default value on top here, and you can remove from where you have OR (||) condition.
 addressCount=0,
  connectionCount=0,
  deviceCount=0,
  activeCount=0

File: console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx (hunk 5 above)
Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/msg-and-iot/utils/ProjectFormatter.tsx:44]: both are same

File: console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx (hunk 2 above)
Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx:29]: The variable is imported in all list component like AddressList ConnectionList. that's why it is exported.
